src/datamanager/core.py

```
@@ -99,7 +140,33 @@ def download_from_r2(client: S3Client, object_key: str, download_path: Path) ->
+    object_key: str, expected_hash: str, output_path: Path, bucket: str | None = None
+) -> bool:
+    """
+    Downloads a file from R2, verifies its hash, and cleans up on failure.
+
+    Returns:
+        True if download and verification succeed, False otherwise.
+    """
+    client = get_r2_client()
+    try:
+        download_from_r2(client, object_key, output_path, bucket)
+    except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)